crates/stackable-operator/src/builder/pod/mod.rs

```diff
@@ -50,9 +51,21 @@ pub enum Error {
 
     #[snafu(display("object is missing key {key:?}"))]
     MissingObjectKey { key: &'static str },
+
+    #[snafu(display("The volume {volume_name:?} clashes with an existing volume of the same name but with different content. \
+        The existing volume is {existing_volume:?}, the new one is {new_volume:?}"))]
+    ClashingVolumeName {
+        volume_name: String,
+        existing_volume: Box<Volume>,
+        new_volume: Box<Volume>,
+    },
```

Comment:
note: Same thought from above applies here as well. I think displaying the volume name should be enough.

```suggestion
    #[snafu(display("tried to add duplicate volume {name:?}"))]
    DuplicateVolume {
        name: String,
    },
```

Reply:
We only have access to the volume, not the volumeMount, so we don't know the mountPath.
We could only say 'the volume "tls" clashes, bruh', which IMHO is too less information. It's not a strong opinion but I think we would be very happy to know what that volume is (emptyDir, configMap or secret-op volume) in case we get such errors.

Reply:
We need to box it to avoid stack out of memory errors, so we put the volumes on the heap. IIRC we already need to do that for other error contents as well.
And yeah, Volume is basically a giant enum

Reply:
So as I said, I would avoid including the volume in the error. More details can be provided via tracing/logging.

> And yeah, Volume is basically a giant enum

That would actually be fine, as enums will only carry one variant at runtime. But `Volume` is actually a struct with 30+ fields (which are mostly empty, but still). I really don't see the need to include this (twice) in the error message.

Reply:
Implemented in a sync meeting